### ISSUE TITLE
Is the valid dataset used?

### ISSUE THREAD

**Issue by saddy001** [junyanz/pytorch-CycleGAN-and-pix2pix#266]
In the docs for pix2pix you recommend a train/valid/test dataset structure. Is the valid set used during training to avoid over-fitting? I could not find any reference in the code. 
If not, what options are there to detect and avoid over-fitting? Can any of the G_* or D_* losses (e.g. G_GAN) be used for this?

**Comment by junyanz** [comment]
It is currently used during the test phase. Sorry for the naming. For pix2pix, L1 loss can be used to detect overfitting.

**Comment by hanrelan** [comment]
Sorry if this is a dumb question but what values would you expect to see for L1 loss if the model is overfitting?

**Comment by junyanz** [comment]
You can measure L1 loss for both training and validation data, and see if there is a big gap.

**Comment by kiyaaabdian** [comment]
hi!
I have some questions about cyclegan and semi supervised cyclegan, can you help me please? 
I create a big dataset with 60k images in it, and I want to use semi supervised cyclegan for it. I don't know how many epoch is enough for training? I'm a little confused about validation set. I also, don't know that we need validation set or not?
tnx for your information

**Comment by junyanz** [comment]
You don't need a validation set for training.

**Comment by kiyaaabdian** [comment]
thank you so much. can you help me please about epoches? I have a paired and unpaired dataset and I don't know how many epoch is enough for supervised section and unsupervised section

**Comment by junyanz** [comment]
This is super hard to decide. It depends on the number of images in your dataset, the complexity of each domain, the difficulty of the task itself, etc. Your best hope is to save a bunch of checkpoints. If you have a validation metric and a proper metric, you can try to pick up the best model. Or you can choose the model based on the manual inspection of the results.

**Comment by Nimbus1997** [comment]
Hello
Similar question here. I am using cycle GAN and want to implement early stopping using the validation set. Is the validation set used in cycleGAN? I am kind of lost on how to import the validation set in the training stage. 
Thanks in advance :)

**Comment by junyanz** [comment]
Sorry for the confusion. There are only train and val in some CycleGAN datasets. So they are not being used in the training test, and only serve as a test set. Feel free to create your own splits.

**Comment by patrickk00** [comment]
hello, i don't undestand why you don't use a validation set during training. 
Also if i need the loss on the test set , where can i find it? it appears only during the training
